### PR TITLE
Move GlobFilter into a separate header/unit

### PR DESCRIPTION
**Pull request by jasnell**
Addresses a long standing todo in workerd where we duplicate GlobFilter there.

### CODE REVIEW

File: c++/src/kj/test-test.c++ [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 #include "common.h"
 #include "test.h"
+#include "glob-filter.h"
```
Comment (kentonv) [inline c++/src/kj/test-test.c++:24]: Should the tests be moved into a `glob-filter-test.c++`?

**Review by kentonv — COMMENTED** [review]
I think you need to update the cmake files too.

**Comment by jasnell** [comment]
> I think you need to update the cmake files too.

Hmm... added but obviously there's still something wrong. Not quite sure what. cmake builds successfully locally but I'm obviously missing something

File: c++/Makefile.am [hunk 2]
```diff
@@ -274,6 +275,7 @@ libkj_la_SOURCES=                                              \
   src/kj/string.c++                                            \
   src/kj/string-tree.c++                                       \
   src/kj/source-location.c++                                   \
+  src/kj/glob-filter.c++
```
Comment (kentonv) [inline c++/Makefile.am:278]: You're missing the trailing backslash on this line.

Comment (jasnell) [reply c++/Makefile.am]: doh! lol sigh.

**Review by kentonv — APPROVED** [review]

**Pull request merged**